assignment/assignment/Scene/Join/JoinNameViewController.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  JoinNameViewController.swift
+//  assignment
+//
+//  Created by 이경민 on 2022/04/06.
+//
+
+import UIKit
+
+final class JoinNameViewController: UIViewController {
+    
+    // MARK: - IBOutlet
+    
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var nextButton: UIButton!
+    
+    // MARK: - LifeCycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUI()
+        setTextField()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        removeTextField()
+    }
+    
+    // MARK: - Function
+    
+    private func setUI() {
+        setButtonUI()
+    }
+    
+    private func setButtonUI() {
+        nextButton.layer.cornerRadius = 15
+    }
+    
+    private func setTextField() {
+        nameTextField.addTarget(self, action: #selector(textFieldDidChange(_:)), for: .editingChanged)
+    }
+    
+    private func removeTextField() {
+        nameTextField.text?.removeAll()
+    }
+    
+    // MARK: - objc function
+    
+    @objc func textFieldDidChange(_ sender: UITextField) {
+        nextButton.isEnabled = nameTextField.hasText
+    }
+    
+    // MARK: - IBAction
+    
+    @IBAction func backButtonDidTap(_ sender: Any) {
+        self.navigationController?.popViewController(animated: true)
+    }
+    
+    @IBAction func nextButtonDidTap(_ sender: Any) {
+        guard let passwordVC = self.storyboard?.instantiateViewController(withIdentifier: "JoinPasswordViewController") as? JoinPasswordViewController else { return }
+        passwordVC.userName = nameTextField.text ?? ""
```

Comment:
```suggestion
        guard let name = nameTextField.text else { return }
        passwordVC.userName = name
```

이런식으로 guard let 구문을 활용해서 옵셔널 처리를 할 수 있을 것 같아요 !!